web/ui/dashboard/src/app/private/components/create-source/create-source.component.ts

```diff
@@ -85,7 +87,7 @@ export class CreateSourceComponent implements OnInit {
 		try {
 			const response = this.action === 'update' ? await this.createSourceService.updateSource({ data: sourceData, id: this.sourceId }) : await this.createSourceService.createSource({ sourceData });
 			this.isloading = false;
-			this.onAction.emit(response.data);
+			this.onAction.emit({ action: this.action, data: response.data });
```

Comment:
This change here, please search everywhere we're using this component and reflect this change